File: docs/gitbook/public/examples/signgate-allow-flow.mjs
import { readFile } from "node:fs/promises";

const baseUrl = process.env.SIGNGATE_BASE_URL;
const apiKey = process.env.SIGNGATE_API_KEY;
if (!baseUrl || !apiKey) throw new Error("Missing SignGate environment variables");

const request = JSON.parse(await readFile("docs/gitbook/public/assets/decision-request.json", "utf8"));
const post = async (path, body) => {
  const response = await fetch(baseUrl + path, {
    method: "POST",
    headers: { authorization: "Bearer " + apiKey, "content-type": "application/json" },
    body: JSON.stringify(body)
  });
  const payload = await response.json().catch(() => null);
  if (!response.ok || !payload) throw new Error("SignGate " + response.status);
  return payload;
};

const decision = await post("/v1/decisions", request);
if (decision.decision !== "ALLOW") throw new Error("Execution stopped: " + decision.decision);
if (JSON.stringify(decision.bound_action) !== JSON.stringify(request.action)) {
  throw new Error("Execution stopped: bound action mismatch");
}

const consumed = await post("/v1/decisions/" + decision.decision_id + "/consume", {
  contract_version: "0.1",
  organization_id: request.organization_id,
  action_fingerprint: decision.action_fingerprint,
  policy_version: decision.policy_version,
  execution_attempt_id: "exec_preview_001"
});
if (!consumed.receipt?.consume_receipt_id) throw new Error("Execution stopped: invalid receipt");
console.log("Authorized for the exact bound action", consumed.receipt);

File: docs/gitbook/public/examples/signgate_allow_flow.py
import json
import os
import urllib.error
import urllib.request

BASE_URL = os.environ["SIGNGATE_BASE_URL"]
API_KEY = os.environ["SIGNGATE_API_KEY"]

with open("docs/gitbook/public/assets/decision-request.json", encoding="utf-8") as source:
    decision_request = json.load(source)

def post(path, body):
    request = urllib.request.Request(
        BASE_URL + path,
        data=json.dumps(body).encode("utf-8"),
        headers={"Authorization": f"Bearer {API_KEY}", "Content-Type": "application/json"},
        method="POST",
    )
    try:
        with urllib.request.urlopen(request, timeout=30) as response:
            return json.load(response)
    except (urllib.error.HTTPError, urllib.error.URLError) as exc:
        raise RuntimeError("Execution stopped: SignGate request failed") from exc

decision = post("/v1/decisions", decision_request)
if decision.get("decision") != "ALLOW":
    raise RuntimeError(f"Execution stopped: {decision.get('decision', 'ERROR')}")
if decision.get("bound_action") != decision_request["action"]:
    raise RuntimeError("Execution stopped: bound action mismatch")

consumed = post(f"/v1/decisions/{decision['decision_id']}/consume", {
    "contract_version": "0.1",
    "organization_id": decision_request["organization_id"],
    "action_fingerprint": decision["action_fingerprint"],
    "policy_version": decision["policy_version"],
    "execution_attempt_id": "exec_preview_001",
})
if not consumed.get("receipt", {}).get("consume_receipt_id"):
    raise RuntimeError("Execution stopped: invalid receipt")
print("Authorized for the exact bound action", consumed["receipt"])
